### PR TITLE
Zipcode restriction

### PR DESCRIPTION
**Pull request by dbaq**
In this PR, I've added two pages to do the zip code restriction #18 

The first iteration contains:
- a map (using leaflet and google maps tiles) via bower, I retrieve the user location and center the map on the user if we got the location
- if the zipcode enters starts by ```94```, the zip code is considered like available (i.e. 94117 good, 54312 not good)
- if it is a valid zipcode, we redirect to the buyer page
- if it is an invalid zipcode, we redirect to a page where the user can enter an email address (I am not really happy with the content of the page, but we can improve it later. The text needs to be proofread and improved though)

What needs to be done in the future:
- after the location is retrieved, we need to call the back end to retrieve the zip code from the gps coordinates
- a call to the server with the zipcode entered to check the real availability
- a call to the server with the zipcode + email address to notify the user once the app is available in its area

So if you are guys are ok, I would like to start the review and merge the PR (once reviewed), we can improve the content and the API calls later.

@valdrinkoshi, @marcandreoli, feedback highly appreciated, thanks :)  

screen shot 2015-06-09 at 9 44 36 pm
screen shot 2015-06-09 at 9 44 22 pm

### CODE REVIEW

**Comment by marcandreoli** [comment]
Thanks @dbaq , I like the first iteration a lot. If this is the new style guide, it looks good to me.

My comments:

Not all zip codes starting with ’94’ should be accepted at first. Only the following: 94114, 94131, 94110, 94102, 94103, 95117.

For the second page, I assume we have already stored the user’s zip code so don’t need to ask it again. I propose the following text: 

“"
Sorry, your area is not yet supported. (bold)

HomeCooked is growing fast. Please enter you email address and we will notify you when your area is available. (normal font)

Email Address (input field)

Get Notified (button)
""

After the user clicks on “get notified”, he can get a “thank you, we will be in touch shortly” pop-up.

@jonbwhite  Take a look at the API calls referenced in the previous message and share any feedback.

File: app/templates/zipcode/form.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<ion-view class="zipcode-restriction-page">
```
Comment (valdrinkoshi) [inline app/templates/zipcode/form.html]: shall we have a nav button to go back/open the menu here? right now we can reach the page only by url

**Comment by valdrinkoshi** [comment]
Nice stuff @dbaq!
It's not clear to me how the user arrives to this page (so far is reachable only by url).
@marcandreoli, so only users from a certain zipcode can use the app? or can it be that someone from Redwood city can still order food in one of the valid zipcodes?

**Comment by valdrinkoshi** [comment]
Ok, now i see, we get to this page if we are not logged in. There is another fix to be done in the menu.controller.js then, to be redirected to this screen if not logged in (in the interceptor of the state change). I'll fix that now

**Comment by valdrinkoshi** [comment]
@dbaq please check my last 3 commits, right now we send the user to validate-zip page only once. If he inputs a good zipcode, we store it in the local cache and next times he'll be sent to buyer's page

**Comment by dbaq** [comment]
You the best @valdrinkoshi, this is what I wanted to do today :) I will review your commits during the day.

**Comment by dbaq** [comment]
All good.

@marcandreoli, I will update the text. Also, after the email is registered I already made a small alert.

I need title + text for this modal too.

**Comment by dbaq** [comment]
@valdrinkoshi I have updated the branch to match @marcandreoli's comments.

I also add a commit to remove the zipcode key in the storage on logout. 

I let you check and merge.

**Comment by valdrinkoshi** [comment]
@dbaq all looks good, I committed a change on the way we were caching things, please have a look and let me know if all looks good.
Right now everytime the user logs out, we send him to the zipcode validation, since the cache with all the data is emptied.
I was thinking we should have the Sign in button on that screen too, something like this
screen shot 2015-06-10 at 11 40 41 pm

What do you think? If that looks good, @dbaq this would need to copy/paste the modal code of the menu.controller into zipcoderestriction.controller which is not great. We can create a ModalService that handles all the modals shared across the app. We can do this in a separate PR or in this one, your call

**Comment by dbaq** [comment]
Your commits looks good to me. However, I am not sure the sign in link on the zip code restriction is a good idea:

- you see this page only if you are not logged in and if you never entered your zip code, which is supposed to happen only once. Nobody logs out from applications nowadays, when was the last time you logged out from an app? (fb, twitter, instagram, etc)
- I believe it adds some distractions on the page, the page has one goal, to let the user enters 5 digits and that's all. The benefit of doing the signing here to skip the zip code validation looks like overkill to me).

So I won't put a link for signin there.

**Comment by valdrinkoshi** [comment]
If you login we clear the cache from all user data (Zipcode included), so you'll see again the validate Zipcode screen.
I don't think we should store the Zip-code on logout.
(Ps: I logout from FB many times, eg when I'm in a new device or in incognito mode :p )

**Comment by valdrinkoshi** [comment]
Sorry I meant "if you logout"

**Comment by valdrinkoshi** [comment]
Anyways, I'll merge this and we can discuss about that and have a separate PR

**Comment by dbaq** [comment]
Ok thanks.

**Pull request merged**